cpp/util/ETSWrapper.h

```diff
@@ -0,0 +1,106 @@
+#ifndef ETSWRAPPER_H
+#define ETSWRAPPER_H
+
+#include <tbb/tbb.h>
+
+namespace freud { namespace util {
+
+//! Wrapper classfor enumerable_thread_specific<T*>
+/*! It is expected that default value for T is 0.
+ */
+template<typename T> class ETSArrayWrapper
+{
+public:
+    //! Default constructor
+    ETSArrayWrapper() : m_size(0), array(tbb::enumerable_thread_specific<T*>([]() { return nullptr; })) {}
+
+    //! Constructor with specific size for thread local arrays
+    /*! \param size Size of the thread local arrays
+     */
+    ETSArrayWrapper(unsigned int size)
+        : m_size(size),
+          array(tbb::enumerable_thread_specific<T*>([this]() { return makeNewEmptyArray(m_size); }))
+    {}
+
+    //! Destructor
+    ~ETSArrayWrapper()
+    {
+        deleteArray();
+    }
+
+    //! Update size of the thread local arrays
+    /*! \param size New size of the thread local arrays
+     */
+    void updateSize(unsigned int size)
```

Comment:
Can we rename this to `resize`? It fits better with other C++ methods for doing essentially the same thing.